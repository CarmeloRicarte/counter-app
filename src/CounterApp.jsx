import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => setCounter(counter + 1);
  const handleSubstract = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>
      <button aria-label="test-increment" onClick={handleAdd}>
        +1
      </button>
      <button aria-label="test-decrement" onClick={handleSubstract}>
        -1
      </button>
      <button aria-label="test-reset" onClick={handleReset}>
        Reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
