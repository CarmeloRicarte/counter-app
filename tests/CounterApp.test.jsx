import { fireEvent, render, screen } from "@testing-library/react";
import CounterApp from "../src/CounterApp";

describe("Pruebas sobre <CounterApp />", () => {
  const initialValue = 100;
  test("debe hacer match con el snapshot", () => {
    const { container } = render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  test("debe mostrar el valor inicial de 100 pasado como prop", () => {
    render(<CounterApp value={initialValue} />);
    expect(screen.getByText(initialValue)).toBeTruthy();
  });

  test("debe de incrementar con el botón +1", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByRole("button", { name: "test-increment" })); // simulate click in button
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      "101"
    );
  });

  test("debe de decrementar con el botón -1", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByRole("button", { name: "test-decrement" })); // simulate click in button
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain("99");
  });

  test("debe de funcionar el botón Reset", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByRole("button", { name: "test-increment" }));
    fireEvent.click(screen.getByRole("button", { name: "test-increment" }));
    fireEvent.click(screen.getByRole("button", { name: "test-increment" }));
    fireEvent.click(screen.getByRole("button", { name: "test-reset" }));
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      "100"
    );
  });
});
