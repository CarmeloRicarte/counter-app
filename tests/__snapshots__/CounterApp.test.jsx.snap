// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <CounterApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    aria-label="test-increment"
  >
    +1
  </button>
  <button
    aria-label="test-decrement"
  >
    -1
  </button>
  <button
    aria-label="test-reset"
  >
    Reset
  </button>
</div>
`;
